src/System.IO.Pipelines.Extensions/StreamExtensions.cs

```diff
@@ -63,16 +63,17 @@ public static IPipeReader AsPipelineReader(this Stream stream, CancellationToken
         /// <param name="stream"></param>
         /// <param name="writer"></param>
         /// <returns></returns>
-        public static Task CopyToAsync(this Stream stream, IPipeWriter writer)
+        public static Task CopyToAsync(this Stream stream, IPipeWriter writer, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return stream.CopyToAsync(new PipelineWriterStream(writer));
+            // 81920 is the default bufferSize, there is not stream.CopyToAsync overload that takes only a cancellationToken
+            return stream.CopyToAsync(new PipelineWriterStream(writer, cancellationToken), bufferSize: 81920, cancellationToken: cancellationToken);
```

Comment:
Because streams are awful and the one passed to CopyToAsync is only used up until write is called internally. Then one passed to PipelineWriterStream is so you can actually cancel the write operation with the cancellation we pass to CopyToAsync